### PR TITLE
rgp: confirmEntry bugfix and style edits

### PR DESCRIPTION
**Pull request by yufengwng**
A couple style fixes for RGP. See comments for more details.

### CODE REVIEW

File: ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss [hunk 1]
```diff
@@ -213,14 +213,14 @@
   margin-bottom: 10px;
   padding: 0.4375em 0;
 
-  border-bottom: 1px solid rgba(0, 0, 0, 0.42);
+  // border-bottom: 1px solid rgba(0, 0, 0, 0.42);
```
Comment (yufengwng) [inline ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss:216]: Not sure what changed recently in Angular side, but for RGP we have "double underline", so removing this bottom border.

Before:
rgp_bug_underline_before

After:
rgp_bug_underline_after

File: ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss [hunk 2]
```diff
@@ -229,9 +229,9 @@ ddp-activity-dropdown-picklist-question select {
   border: none;
   border-bottom: 1px solid rgba(0, 0, 0, 0.42);
 
-  &:disabled {
-    border-bottom-style: dotted;
-  }
+  // &:disabled {
+  //   border-bottom-style: dotted;
+  // }
```
Comment (yufengwng) [inline ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss:234]: I noticed an extra dotted line when in read-only mode, so removing this.

rgp_bug_dotted

File: ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss [hunk 3]
```diff
@@ -377,3 +377,44 @@ ddp-activity-dropdown-picklist-question select {
     padding: 0;
   }
 }
+
+/**
+ * Enrollment nested activities
+ */
+.instance {
+  .instance__divider {
```
Comment (yufengwng) [inline ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss:385]: This CSS block is to style the new "divider" for "embedded nested activities". The `.instance:first-child` selector below is to hide the divider for the first instance.

Before:
rgp_bug_divider_before

After:
rgp_bug_divider_after

File: ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss [hunk 4]
```diff
@@ -377,3 +377,44 @@ ddp-activity-dropdown-picklist-question select {
...
+  }
+}
+
+.instance:first-child .instance__divider {
+  display: none;
+}
+
+.instance:first-child .instance__header {
+  display: none;
+}
```
Comment (yufengwng) [inline ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss:407]: This is necessary to hide the header label for the first instance (this is something new for embedded nested activities!).

Before:
rgp_bug_label_before

After:
rgp_bug_label_after

File: ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss [hunk 5]
```diff
@@ -377,3 +377,44 @@ ddp-activity-dropdown-picklist-question select {
...
+
+.add-btn {
```
Comment (yufengwng) [inline ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss:409]: I got pretty far with styling the button, but I think I need help with the padding/spacing...

Before:
rgp_bug_button_before

After:
rgp_bug_button_after

File: ddp-workspace/projects/ddp-sdk/src/lib/services/activity/activityQuestionConverter.service.ts [hunk 6]
```diff
@@ -88,6 +88,10 @@ export class ActivityQuestionConverter {
                     questionBlock.setAnswer(answer, false);
                 } else {
                     questionBlock.answer = valueForQuestion;
+                    if (questionJson.questionType === 'TEXT' && questionJson.confirmEntry) {
+                        const textQuestion = questionBlock as ActivityTextQuestionBlock;
+                        textQuestion.confirmationAnswer = textQuestion.answer;
```
Comment (yufengwng) [inline ddp-workspace/projects/ddp-sdk/src/lib/services/activity/activityQuestionConverter.service.ts:93]: Ran into an issue where the `ActivityMatchValidationRule` will fail whenever I reload the activity page or when we're in read-only mode. I realized that the rule is checking whether `confirmAnswer == answer` but we're not setting `confirmAnswer` anywhere when initializing. I think the json converter here is a good place to do that.

rgp_bug_confirm

File: ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss (hunk 5 above)
Comment (aleksey-kuznetsov) [reply ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss:409]: How about adding this:
```suggestion
.add-btn {
  display: inline-flex;
  align-items: center;
  justify-content: center;
}
```

This should probably fix the issue with text positioning

Comment (yufengwng) [reply ddp-workspace/projects/ddp-rgp/src/styles/_activity.scss:409]: Made a few more style edits to the button. I think it should be good for now.

**Comment by yufengwng** [comment]
@charlottedye @aleksey-kuznetsov Can I get your review on this PR?

**Review by aleksey-kuznetsov — APPROVED** [review]
Seems good! Thanks for your help!

**Pull request merged**